src/openloco/windows/town_list.cpp

```diff
@@ -1,16 +1,1581 @@
+                    interface_skin::image_ids::build_misc_buildings_frame_10,
+                    interface_skin::image_ids::build_misc_buildings_frame_11,
+                    interface_skin::image_ids::build_misc_buildings_frame_12,
+                    interface_skin::image_ids::build_misc_buildings_frame_13,
+                    interface_skin::image_ids::build_misc_buildings_frame_14,
+                    interface_skin::image_ids::build_misc_buildings_frame_15,
+                };
+                uint32_t imageId = skin->img;
+                if (self->current_tab == widx::tab_build_misc_buildings - widx::tab_town_list)
+                    imageId += buildMiscBuildingsImageIds[(self->frame_no / 2) % std::size(buildMiscBuildingsImageIds)];
+                else
+                    imageId += buildMiscBuildingsImageIds[0];
+
+                widget::draw_tab(self, dpi, imageId, widx::tab_build_misc_buildings);
+            }
+        }
+
+        //0x00457F27
+        static void switchTab(window* self, widget_index widgetIndex)
+        {
+            if (input::is_tool_active(self->type, self->number))
+                input::cancel_tool();
+
+            self->current_tab = widgetIndex - widx::tab_town_list;
+            self->frame_no = 0;
+            self->flags &= ~(window_flags::flag_16);
+
+            if (self->viewports[0] != nullptr)
+            {
+                self->viewports[0]->width = 0;
+                self->viewports[0] = nullptr;
+            }
+
+            const auto& tabInfo = tabInformationByTabOffset[widgetIndex - widx::tab_town_list];
+
+            self->enabled_widgets = tabInfo.enabledWidgets;
+            self->holdable_widgets = 0;
+            self->event_handlers = tabInfo.events;
+            self->activated_widgets = 0;
+            self->widgets = tabInfo.widgets;
+
+            self->invalidate();
+
+            if (self->current_tab == widx::tab_town_list - widx::tab_town_list)
+                town_list::tabReset(self);
+            if (self->current_tab == widx::tab_build_town - widx::tab_town_list)
+                build_towns::tabReset(self);
+            if (self->current_tab == widx::tab_build_buildings - widx::tab_town_list || self->current_tab == widx::tab_build_misc_buildings - widx::tab_town_list)
+                build_buildings::tabReset(self);
+
+            self->call_on_resize();
+            self->call_prepare_draw();
+            self->init_scroll_widgets();
+            self->invalidate();
+            self->moveInsideScreenEdges();
+        }
+
+        // 0x00499DDE
+        static void refreshTownList(window* self)
+        {
+            self->row_count = 0;
+
+            for (auto& town : townmgr::towns())
+            {
+                if (town.empty())
+                    continue;
+
+                town.flags &= ~town_flags::sorted;
+            }
+        }
+
+        // 0x00468FD3
+        static void showGridlines()
```

Comment:
This is a function that is already implemented for the industry window. Move this into the main windowmanger file as this is used by a number of windows.